src/envoy/utils/grpc_transport.cc

```diff
@@ -38,12 +38,20 @@ GrpcTransport<RequestType, ResponseType>::GrpcTransport(
     : async_client_(std::move(async_client)),
       response_(response),
       serialized_forward_attributes_(serialized_forward_attributes),
-      on_done_(on_done),
-      request_(async_client_->send(
-          descriptor(), request, *this, parent_span,
-          absl::optional<std::chrono::milliseconds>(kGrpcRequestTimeoutMs))) {
+      on_done_(on_done) {
   ENVOY_LOG(debug, "Sending {} request: {}", descriptor().name(),
             request.DebugString());
+  Envoy::Http::AsyncClient::RequestOptions options;
+  options.setTimeout(kGrpcRequestTimeoutMs);
+  Protobuf::RepeatedPtrField<envoy::api::v2::route::RouteAction::HashPolicy>
+      hash_policy;
+  hash_policy.Add()->mutable_header()->set_header_name(
```

Comment:
what is this trying to do? 
adding HostHeader as a hash policy? why 2 calls?

Reply:
This sets up a hash based on combination of x-istio-attributes and :authority as a fallback. Istio attributes carry source uid. Unfortunately, source IP hash policy won't work since downstream address is not set for async client.